pkg/frontend/transport/handler_test.go

```diff
@@ -97,3 +98,47 @@ func TestHandler_ServeHTTP(t *testing.T) {
 		})
 	}
 }
+
+func TestHandler_FailedRoundTrip(t *testing.T) {
+	for _, test := range []struct {
+		name            string
+		cfg             HandlerConfig
+		expectedMetrics int
+	}{
+		{
+			name:            "Failed round trip with context cancelled",
+			cfg:             HandlerConfig{QueryStatsEnabled: true},
```

Comment:
What difference does this parameter make? I tried disabling it, the test still passes.